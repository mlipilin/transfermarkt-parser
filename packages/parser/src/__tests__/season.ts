import { season } from '../index'

const timeout = Number(process.env.JEST_ASYNC_TIMEOUT)

describe('list', () => {
  test(
    'season.list("GB1") -> list of seasons for EPL',
    async () => {
      await expect(season.list('GB1')).resolves.toMatchSnapshot()
    },
    timeout
  )
  test(
    'season.list("wrongCompetitionId") -> []',
    async () => {
      await expect(season.list('wrongCompetitionId')).resolves.toMatchSnapshot()
    },
    timeout
  )
  test(
    'season.list() -> []',
    async () => {
      // @ts-ignore
      await expect(season.list()).resolves.toMatchSnapshot()
    },
    timeout
  )
})
