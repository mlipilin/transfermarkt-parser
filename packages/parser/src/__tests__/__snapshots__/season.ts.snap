// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list season.list("GB1") -> list of seasons for EPL 1`] = `
[
  {
    "competitionId": "GB1",
    "id": "2022",
    "title": "22/23",
  },
  {
    "competitionId": "GB1",
    "id": "2021",
    "title": "21/22",
  },
  {
    "competitionId": "GB1",
    "id": "2020",
    "title": "20/21",
  },
  {
    "competitionId": "GB1",
    "id": "2019",
    "title": "19/20",
  },
  {
    "competitionId": "GB1",
    "id": "2018",
    "title": "18/19",
  },
  {
    "competitionId": "GB1",
    "id": "2017",
    "title": "17/18",
  },
  {
    "competitionId": "GB1",
    "id": "2016",
    "title": "16/17",
  },
  {
    "competitionId": "GB1",
    "id": "2015",
    "title": "15/16",
  },
  {
    "competitionId": "GB1",
    "id": "2014",
    "title": "14/15",
  },
  {
    "competitionId": "GB1",
    "id": "2013",
    "title": "13/14",
  },
  {
    "competitionId": "GB1",
    "id": "2012",
    "title": "12/13",
  },
  {
    "competitionId": "GB1",
    "id": "2011",
    "title": "11/12",
  },
  {
    "competitionId": "GB1",
    "id": "2010",
    "title": "10/11",
  },
  {
    "competitionId": "GB1",
    "id": "2009",
    "title": "09/10",
  },
  {
    "competitionId": "GB1",
    "id": "2008",
    "title": "08/09",
  },
  {
    "competitionId": "GB1",
    "id": "2007",
    "title": "07/08",
  },
  {
    "competitionId": "GB1",
    "id": "2006",
    "title": "06/07",
  },
  {
    "competitionId": "GB1",
    "id": "2005",
    "title": "05/06",
  },
  {
    "competitionId": "GB1",
    "id": "2004",
    "title": "04/05",
  },
  {
    "competitionId": "GB1",
    "id": "2003",
    "title": "03/04",
  },
  {
    "competitionId": "GB1",
    "id": "2002",
    "title": "02/03",
  },
  {
    "competitionId": "GB1",
    "id": "2001",
    "title": "01/02",
  },
  {
    "competitionId": "GB1",
    "id": "2000",
    "title": "00/01",
  },
  {
    "competitionId": "GB1",
    "id": "1999",
    "title": "99/00",
  },
  {
    "competitionId": "GB1",
    "id": "1998",
    "title": "98/99",
  },
  {
    "competitionId": "GB1",
    "id": "1997",
    "title": "97/98",
  },
  {
    "competitionId": "GB1",
    "id": "1996",
    "title": "96/97",
  },
  {
    "competitionId": "GB1",
    "id": "1995",
    "title": "95/96",
  },
  {
    "competitionId": "GB1",
    "id": "1994",
    "title": "94/95",
  },
  {
    "competitionId": "GB1",
    "id": "1993",
    "title": "93/94",
  },
  {
    "competitionId": "GB1",
    "id": "1992",
    "title": "92/93",
  },
]
`;

exports[`list season.list("wrongCompetitionId") -> [] 1`] = `[]`;

exports[`list season.list() -> [] 1`] = `[]`;
